registry/raw.go

```diff
@@ -34,8 +34,28 @@ func (j *rawJob) GetBSON() (interface{}, error) { // Get ~= Marshal
 	return j.job, nil
 }
 
-func (j *rawJob) UnmasrhalJSON(data []byte) error { j.Body = data; return nil }
+func (j *rawJob) UnmarshalYAML(um func(interface{}) error) error {
+	factory, err := GetJobFactory(j.Type)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	job := factory()
+
+	err = um(job)
+	if err != nil {
+		return err
+	}
+
+	j.job = job
 
+	j.Body, err = amboy.ConvertTo(amboy.YAML, job)
```

Comment:
Shouldn't this be in the MarshalYAML() function?

Reply:
Doing this makes the tests pass, though your right it shouldn't be needed.

Without this I think you couldn't roundtrip and then move to a different serialization method, because the yaml library is ~bonkers